### ISSUE TITLE
Crashing on Export of Big Dataset

### ISSUE THREAD

**Issue by ljk1291** [OllegK/InstagramHelper#25]
Hi,
Im trying to export over 300k likes but each time I click "Stop it" the extension crashes before loading the data or when I click export the extension crashes and all the data is lost. I was wondering if the JSON is saved somewhere by the extension? Or if there is any way to collect the data after a crash?

**Comment by OllegK** [comment]
Hi,
What does it mean the extension crashes, does the page becomes unresponsive? On exporting the large datasets, the application could become unresponsive for a long time, some people considering that as a crash. 
The collected info is in the memory only, so no way to get the date after a tab is closed.

**Comment by ljk1291** [comment]
No, its not unresponsive. I get the message "Instagram Helper has crashed click this balloon to reload the extension".

I'll try to reproduce it and see if I can get more data from the console

**Comment by ljk1291** [comment]
Its seems to be crashing because chrome is running out of memory. I had the dev tools open while exporting and it "paused before a potential out of memory crash". This chrome instance seemed to be using 2GB of memory. The memory Tab in dev tools showed the extension using 2056 out of 2104MB, not sure why only 2GB is allocated to the extension. And Im also not sure what is causing that much memory to be used on export. I tried to take a memory snapshot, but chrome decided to crash while it was running... 

This is on a dedicated server with 64GB of RAM, so memory shouldnt be an issue

**Comment by OllegK** [comment]
Thank you for this investigation. How many records did you get in the generated list? I will try to reproduce it. I am using js-xlsx for export; I will update to the latest version also and also validate if I use it correctly.

**Comment by ljk1291** [comment]
I think it was around 220K records. I did some more debugging. I ran it over night on a big instagrammer. I got around 450K records and hit "Stop It" this morning. 

The debugger paused on this part this time with the same error "paused before a potential out of memory crash":
```
go.prototype.walk = function(e) { e = {pic: "https://scontent-prg1-1.cdninstagram.com/vp/432ee6…107217_2093192307404848_7902373342791386735_n.jpg", url: "https://www.instagram.com/p/BrK..
        for (var t = Object.keys(e), n = 0; n < t.length; n++) t = (3) ["id", "pic", "url"], n = 1 
            M(e, t[n], e[t[n]])
    }
```
It looks like it is using a ton of memory to collect the pictures. Are they all loaded at once?
This time chrome went up to 3.8GB RAM before crashing :D

**Comment by OllegK** [comment]
I will check it, but it seems it is not because of the loaded pictures, the array itself could occupy a lot of memory, a record contains long values, and as the array is bound to UI component, and VUE.JS adds the additional properties/method that also occupies memory.

**Comment by influencerdeal** [comment]
also experienced the same issue for large accounts on a slower / older machine
